simulation-construction-set/falling-brick/src/main/java/us/ihmc/fallingBrick/FallingBrickRobot.java

```diff
@@ -0,0 +1,204 @@
+       * associated with the joint. The second parameter "new Vector3d(0.0, 0.0, 0.0)" defines the offset
+       * of this joint from the previous joint. The third parameter "this" refers to the robot itself.
+       */
+      floatingJoint = new FloatingJoint("base", new Vector3D(0.0, 0.0, 0.0), this);
+      Link link1 = base("base", YoAppearance.Purple());
+      // Sets the link created previously as the link for this joint
+      floatingJoint.setLink(link1);
+
+      // Add a RootJoint to the robot
+      /*
+       * This line adds floatingJoint as a rootJoint of the robot. In order to be part of a robot, a joint
+       * must either be added as a root joint, or be attached to a parent joint. This ensures the tree
+       * structure (forest structure if there are multiple root joints) of the robot.
+       */
+      addRootJoint(floatingJoint);
+
+      // add ground contact points to the brick
+      GroundContactPoint gc1 = new GroundContactPoint("gc1", new Vector3D(BASE_L / 2.0, BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc1);
+      GroundContactPoint gc2 = new GroundContactPoint("gc2", new Vector3D(BASE_L / 2.0, -BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc2);
+      GroundContactPoint gc3 = new GroundContactPoint("gc3", new Vector3D(-BASE_L / 2.0, BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc3);
+      GroundContactPoint gc4 = new GroundContactPoint("gc4", new Vector3D(-BASE_L / 2.0, -BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc4);
+
+      GroundContactPoint gc5 = new GroundContactPoint("gc5", new Vector3D(BASE_L / 2.0, BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc5);
+      GroundContactPoint gc6 = new GroundContactPoint("gc6", new Vector3D(BASE_L / 2.0, -BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc6);
+      GroundContactPoint gc7 = new GroundContactPoint("gc7", new Vector3D(-BASE_L / 2.0, BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc7);
+      GroundContactPoint gc8 = new GroundContactPoint("gc8", new Vector3D(-BASE_L / 2.0, -BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc8);
+
+      GroundContactPoint gc9 = new GroundContactPoint("gc9", new Vector3D(0.0, 0.0, BASE_H / 2.0 + BASE_H), this);
+      floatingJoint.addGroundContactPoint(gc9);
+      GroundContactPoint gc10 = new GroundContactPoint("gc10", new Vector3D(0.0, 0.0, -BASE_H / 2.0 - BASE_H), this);
+      floatingJoint.addGroundContactPoint(gc10);
+
+      this.setController(this); // tells the simulator to call the local doControl() method
+
+      // instantiate ground contact model
+      GroundContactModel groundModel = new LinearGroundContactModel(this, 1422, 150.6, 50.0, 1000.0, getRobotsYoVariableRegistry());
```

Comment:
Maybe a piece of comment for the ground contact model too.
FYI: the model is pretty much like a controller that is called every simulation tick and which job is to detect contact points colliding with the ground. When a contact point collides with the ground, the model then computes the force to be applied on the robot at the contact point to resolve collision. The `LinearGroundContactModel` uses a spring-damper based strategy to compute the force to be applied, the stiffness and damping parameters are axis dependent: 1 set of stiffness+damping values is used for computing the force along the contact normal, 1 other set of stiffness+damping values is used for computing the force tangent to the contact: the friction force. The first pair of values is for the stiffness & damping for computing the friction force, the second pair of values is for the stiffness & damping used to compute the normal force. It is also worth noting that internally, the ground contact model uses a default coefficient of friction that is used to ensure that the ground reaction force remains within a friction cone.